crates/orchestrator/src/jobs/da_job/mod.rs

```diff
@@ -616,4 +673,55 @@ pub mod test {
         new_hex_chars = new_hex_chars.trim_start_matches('0').to_string();
         if new_hex_chars.is_empty() { "0x0".to_string() } else { format!("0x{}", new_hex_chars) }
     }
+
+    fn create_state_diff(
+        nonces: Vec<(Felt, Felt)>,
+        storage_diffs: Vec<(Felt, Vec<(Felt, Felt)>)>,
+        deployed_contracts: Vec<(Felt, Vec<u8>)>,
+    ) -> StateDiff {
+        StateDiff {
+            nonces: nonces.into_iter().map(|(addr, nonce)| NonceUpdate { contract_address: addr, nonce }).collect(),
+            storage_diffs: storage_diffs
+                .into_iter()
+                .map(|(addr, entries)| ContractStorageDiffItem {
+                    address: addr,
+                    storage_entries: entries.into_iter().map(|(key, value)| StorageEntry { key, value }).collect(),
+                })
+                .collect(),
+            deprecated_declared_classes: vec![],
+            declared_classes: vec![],
+            deployed_contracts: deployed_contracts
+                .into_iter()
+                .map(|(addr, _code)| DeployedContractItem { address: addr, class_hash: Default::default() })
+                .collect(),
+            replaced_classes: vec![],
+        }
+    }
+
+    fn verify_unique_addresses(state_diff: &StateDiff, expected_count: usize) {
+        let unique_addresses: HashSet<_> = state_diff.storage_diffs.iter().map(|item| &item.address).collect();
+
+        assert_eq!(unique_addresses.len(), state_diff.storage_diffs.len(), "Storage diffs contain duplicate addresses");
+        assert_eq!(unique_addresses.len(), expected_count, "Unexpected number of storage diffs");
+    }
+
+    fn verify_storage_preservation(state_diff: &StateDiff, initial_storage: &Vec<ContractStorageDiffItem>) {
+        for orig_storage in initial_storage {
+            if let Some(current_storage) =
+                state_diff.storage_diffs.iter().find(|item| item.address == orig_storage.address)
+            {
+                assert_eq!(
+                    orig_storage.storage_entries, current_storage.storage_entries,
+                    "Storage entries changed unexpectedly"
+                );
+            }
+        }
+    }
+
+    fn verify_addresses_have_storage_diffs(state_diff: &StateDiff) -> bool {
```

Comment:
this function is not being called in the test

Reply:
resolved

Reply:
did you add it? I might be looking at the wrong place

Reply:
this is being called here : 

https://github.com/madara-alliance/madara-orchestrator/pull/176/files#diff-01a67dcc79495a881151d118f8080aa28ead03df572a5d8e16a8f6eebc326225R622